### PR TITLE
[l2-load-balancer] add externalTrafficPolicy option

### PR DESCRIPTION
**Pull request by Beastlex**
## Description
Add externalTrafficPolicy option to L2LoadBalancer custom resource.

## Why do we need it, and what problem does it solve?
We need to be able to define how to route external traffic to node-local or cluster-wide endpoints. (Missed Service spec).

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: l2lLoadBalancer
type: fix
summary: Missed externalTrafficPolicy option for L2LoadBalancer.
```

### CODE REVIEW

File: ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balncer.yaml [hunk 1]
```diff
@@ -20,6 +20,11 @@ spec:
                   sourceRanges:
                     description: |
                       Список разрешённых диапазонов клиентских IP-адресов для взаимодействия с сервисом.
+                  externalTrafficPolicy:
+                    description: |
+                      Описывает каким образом LoadBalancer отправляет трафик к конечным подам:
+                      * Local (по умолчанию) - трафик направляется к подам, расположенным на той же ноде, что и L2 LoadBalancer speaker
+                      * Cluster - трафик может быть направлени на любой под в кластере.
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balncer.yaml:27]:
```suggestion
                      * Local (по умолчанию) — трафик направляется к подам, расположенным на той же ноде, что и L2 LoadBalancer speaker
                      * Cluster — трафик может быть направлени на любой под в кластере.
```

File: ee/modules/381-l2-load-balancer/docs/EXAMPLES.md [hunk 2]
```diff
@@ -58,7 +58,17 @@ title: "The l2-load-balancer module: examples"
   d8-l2-load-balancer-nginx-1   LoadBalancer   10.222.91.98    192.168.199.104   8000:30806/TCP   1s
   d8-l2-load-balancer-nginx-2   LoadBalancer   10.222.186.57   192.168.199.105   8000:30272/TCP   1s
   ```
-
+  The assigned IP addresses can also be seen in the Status section of the custom resource L2LoadBalancer:
+  ```bash
+  kubectl describe l2loadbalancers.deckhouse.io test
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/docs/EXAMPLES.md:63]:
```suggestion
  kubectl describe l2loadbalancers.deckhouse.io nginx
```

File: ee/modules/381-l2-load-balancer/docs/EXAMPLES_RU.md [hunk 3]
```diff
@@ -58,6 +58,17 @@ title: "The l2-load-balancer module: примеры"
   d8-l2-load-balancer-nginx-1   LoadBalancer   10.222.91.98    192.168.199.104   8000:30806/TCP   1s
   d8-l2-load-balancer-nginx-2   LoadBalancer   10.222.186.57   192.168.199.105   8000:30272/TCP   1s
   ```
+  
+  Назначенные IP адреса можно также увидеть в разделе Status объекта L2LoadBalacner:
+  ```bash
+  kubectl describe l2loadbalancers.deckhouse.io test
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/docs/EXAMPLES_RU.md:64]:
```suggestion
  kubectl describe l2loadbalancers.deckhouse.io nginx
```

File: ee/modules/381-l2-load-balancer/hooks/handle_load_balancers.go [hunk 4]
```diff
@@ -65,14 +65,20 @@ func applyLoadBalancerFilter(obj *unstructured.Unstructured) (go_hook.FilterResu
 		return nil, err
 	}
 
+	externalTrafficPolicy := "Local"
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/hooks/handle_load_balancers.go:68]: default value is in openapi.

File: ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balncer.yaml [hunk 5]
```diff
@@ -20,6 +20,11 @@ spec:
                   sourceRanges:
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balncer.yaml:20]: pls, configure the externalTrafficPolicy of L2LoadBalancer in ingress-nginx inlet explicitly.

**Review by apolovov — CHANGES_REQUESTED** [review]

File: ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balancer.yaml [hunk 6]
```diff
@@ -20,6 +20,11 @@ spec:
                   sourceRanges:
                     description: |
                       Список разрешённых диапазонов клиентских IP-адресов для взаимодействия с сервисом.
+                  externalTrafficPolicy:
+                    description: |
+                      Описывает каким образом LoadBalancer отправляет трафик к конечным подам:
+                      * Local (по умолчанию) — трафик направляется к подам, расположенным на той же ноде, что и L2 LoadBalancer speaker
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balancer.yaml:26]:
```suggestion
                      * Local (по умолчанию) — трафик направляется к подам, расположенным на той же ноде, что и L2 LoadBalancer speaker.
```

File: ee/modules/381-l2-load-balancer/docs/EXAMPLES.md [hunk 7]
```diff
@@ -58,7 +58,19 @@ title: "The l2-load-balancer module: examples"
   d8-l2-load-balancer-nginx-1   LoadBalancer   10.222.91.98    192.168.199.104   8000:30806/TCP   1s
   d8-l2-load-balancer-nginx-2   LoadBalancer   10.222.186.57   192.168.199.105   8000:30272/TCP   1s
   ```
-
+  
+  The assigned IP addresses can also be seen in the Status section of the custom resource L2LoadBalancer:
```
Comment (apolovov) [inline ee/modules/381-l2-load-balancer/docs/EXAMPLES.md:62]:
```suggestion
  The assigned IP addresses can also be seen in the `status` section of the custom resource L2LoadBalancer:
```

**Review by apolovov — CHANGES_REQUESTED** [review]

**Review by apolovov — APPROVED** [review]

File: ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balancer.yaml [hunk 8]
```diff
@@ -20,6 +20,11 @@ spec:
                   sourceRanges:
                     description: |
                       Список разрешённых диапазонов клиентских IP-адресов для взаимодействия с сервисом.
+                  externalTrafficPolicy:
+                    description: |
+                      Описывает каким образом LoadBalancer отправляет трафик к конечным подам:
+                      * Local (по умолчанию) — трафик направляется к подам, расположенным на той же ноде, что и L2 LoadBalancer speaker.
+                      * Cluster — трафик может быть направлени на любой под в кластере.
```
Comment (z9r5) [inline ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balancer.yaml:27]:
```suggestion
                      Описывает каким образом LoadBalancer отправляет трафик к конечным подам:
                      * `Local` (по умолчанию) — трафик направляется к подам, расположенным на том же узле, что и L2 LoadBalancer speaker.
                      * `Cluster` — трафик может быть направлен на любой под в кластере.
```

File: ee/modules/381-l2-load-balancer/crds/l2-load-balancer.yaml [hunk 9]
```diff
@@ -60,6 +60,13 @@ spec:
                     items:
                       type: string
                       pattern: '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/(3[0-2]|[1-2][0-9]|[0-9]))$'
+                  externalTrafficPolicy:
+                    type: string
+                    x-doc-d8Revision: ee
+                    default: "Local"
+                    enum: ["Local", "Cluster"]
+                    description: |
+                      Denotes if this L2 LoadBalancer desires to route external traffic to node-local or cluster-wide endpoints. There are two available options: Local (default) and Cluster.
```
Comment (z9r5) [inline ee/modules/381-l2-load-balancer/crds/l2-load-balancer.yaml:69]:
```suggestion
                      Denotes if this L2 LoadBalancer desires to route external traffic to node-local or cluster-wide endpoints. There are two available options: Local (default) and Cluster.
                      * `Local` (by default) — traffic is routed to pods located on the same node as L2 LoadBalancer speaker.
                      * `Cluster` — traffic can be routed to any pod in the cluster.
```

File: ee/modules/381-l2-load-balancer/docs/EXAMPLES_RU.md [hunk 10]
```diff
@@ -58,6 +58,18 @@ title: "The l2-load-balancer module: примеры"
   d8-l2-load-balancer-nginx-1   LoadBalancer   10.222.91.98    192.168.199.104   8000:30806/TCP   1s
   d8-l2-load-balancer-nginx-2   LoadBalancer   10.222.186.57   192.168.199.105   8000:30272/TCP   1s
   ```
+  
+  Назначенные IP адреса можно также увидеть в разделе Status объекта L2LoadBalacner:
```
Comment (z9r5) [inline ee/modules/381-l2-load-balancer/docs/EXAMPLES_RU.md:62]:
```suggestion
  Назначенные IP-адреса можно также увидеть в поле `status` ресурса `L2LoadBalancer`:
```

File: ee/modules/381-l2-load-balancer/docs/EXAMPLES.md [hunk 11]
```diff
@@ -58,7 +58,19 @@ title: "The l2-load-balancer module: examples"
   d8-l2-load-balancer-nginx-1   LoadBalancer   10.222.91.98    192.168.199.104   8000:30806/TCP   1s
   d8-l2-load-balancer-nginx-2   LoadBalancer   10.222.186.57   192.168.199.105   8000:30272/TCP   1s
   ```
-
+  
+  The assigned IP addresses can also be seen in the `status` section of the custom resource L2LoadBalancer:
```
Comment (z9r5) [inline ee/modules/381-l2-load-balancer/docs/EXAMPLES.md:62]:
```suggestion
  The assigned IP addresses can also be seen in the `status` section of the custom resource `L2LoadBalancer`:
```

**Review by z9r5 — APPROVED** [review]

File: ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balancer.yaml [hunk 12]
```diff
@@ -20,6 +20,11 @@ spec:
                   sourceRanges:
                     description: |
                       Список разрешённых диапазонов клиентских IP-адресов для взаимодействия с сервисом.
+                  externalTrafficPolicy:
+                    description: |
+                      Описывает каким образом LoadBalancer отправляет трафик к конечным подам:
```
Comment (z9r5) [inline ee/modules/381-l2-load-balancer/crds/doc-ru-l2-load-balancer.yaml:25]:
```suggestion
                      Описывает каким образом балансировщик отправляет трафик к конечным подам:
```

**Pull request merged**